From 0eacd579e22b636ab146aac944a819188d027d1d Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Wed, 30 Jan 2019 13:15:38 -0800
Subject: [PATCH 1003/1200] drivers: i2c: mask read flag in probe_chip

Use of read flag breaks armada 3k platforms, fix with use
of ARCH_OCTEONTX conditional compile.
This is needed for OcteonTX as customer reported failure
with some slave devices.

Change-Id: I2d0c7b860332f626f9c1ebd9628ed9ba92740867
Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
Reviewed-on: https://sj1git1.cavium.com/3046
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
---
 drivers/i2c/i2c-uclass.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/i2c/i2c-uclass.c b/drivers/i2c/i2c-uclass.c
index 08f13db3ba..58a14dbcae 100644
--- a/drivers/i2c/i2c-uclass.c
+++ b/drivers/i2c/i2c-uclass.c
@@ -256,7 +256,9 @@ static int i2c_probe_chip(struct udevice *bus, uint chip_addr,
 	/* Probe with a zero-length message */
 	msg->addr = chip_addr;
 	msg->flags = chip_flags & DM_I2C_CHIP_10BIT ? I2C_M_TEN : 0;
+#ifdef CONFIG_ARCH_OCTEONTX
 	msg->flags |= I2C_M_RD;
+#endif
 	msg->len = 0;
 	msg->buf = NULL;
 
-- 
2.22.0

